modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/portlet/action/GetLayoutReportsIssuesMVCResourceCommand.java

```diff
@@ -160,12 +161,17 @@ protected void doServeResource(
 
 	private String _getCompleteURL(ThemeDisplay themeDisplay) {
 		try {
-			return _portal.getLayoutURL(themeDisplay);
+			String url = _portal.getLayoutURL(themeDisplay);
```

Comment:
We usually inline single use variables like this:

```java
return _http.addParameter(
    _portal.getLayoutURL(themeDisplay), "keepAuditPanelOpen", true);
```

But it's OK anyway. Brian might ask you to rename `url` to `layoutURL` and `currentCompleteURL` respectively, but I am not sure so we could try sending it like this if we finally go for this solution.

Reply:
I divided it in two lines for clarity, happy to inline it ;)

Reply:
**LayoutReportsProductNavigationControlMenuEntry** is a component and as far as I remember, utils classes are not allowed there, is **ParamUtil** an exception to the rule?

Reply:
Yes it is 😂. ParamUtil is a utility class. The utils that are not allowed are those wrapping an OSGi service.